### PR TITLE
feat(vep): run vep in panel pipe

### PR DESCRIPTION
**Pull request by jemten**
### This PR fixes:

- modifies rna version of vep recipe to use different padding for mt
- use vep rna recipe in panel pipe
- part of #1424 

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Recipes/Analysis/Vep.pm [hunk 1]
```diff
@@ -1360,14 +1361,32 @@ sub analysis_vep_rna {
...
+    my $vep_synonyms_file_path = create_vep_synonyms_file(
+        {
+            log          => $log,
+            outfile_path => catfile( $outdir_path_prefix, q{synonyms.tsv} ),
+            version      => $genome_reference_version,
+        }
+    );
+
+    ## Get contigs
+    my %primary_contig = Readonly::Clone %PRIMARY_CONTIG;
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Vep.pm]: I think this section should be 2 get subs in lib/MIP/Contigs.pm instead. One which returns a specific contig(s) and one that returns all but the specifc contig(s). Then this recipe does not have to know anything about %PRIMARY_CONTIG and its structure and you get 2 general and unit tested sub for reuse elsewhere.

File: lib/MIP/Recipes/Analysis/Vep.pm [hunk 2]
```diff
@@ -1401,35 +1428,67 @@ sub analysis_vep_rna {
...
             vep_features_ref       => \@vep_features_ref,
         }
     );
     say {$filehandle} $NEWLINE;
 
+    ## VEP for MT
+    variant_effect_predictor(
+        {
+            assembly               => $assembly_version,
+            buffer_size            => 20_000,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Vep.pm]: Make it into a READONLY constant

File: lib/MIP/Recipes/Pipeline/Analyse_rd_dna_panel.pm [hunk 3]
```diff
@@ -178,8 +178,9 @@ sub pipeline_analyse_rd_dna_panel {
     use MIP::Recipes::Analysis::Sambamba_depth qw{ analysis_sambamba_depth };
     use MIP::Recipes::Analysis::Samtools_merge qw{ analysis_samtools_merge_panel };
     use MIP::Recipes::Analysis::Variant_integrity qw{ analysis_variant_integrity };
+    use MIP::Recipes::Analysis::Vep qw{ analysis_vep_rna };
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Analyse_rd_dna_panel.pm]: This intuitively feels strange with `rna` in the Analyse_rd_dna_panel. Could we rename the analysis recipe to something more generic? I know that we usually use pipeline_type in the name, but it feels odd. Maybe we could switch the:
analysis_vep -> analysis_vep_wgs (or analysis_vep_per_contig)
analysis_vep_rna -> analysis_vep
What do you think?

**Comment by henrikstranneheim** [comment]
Seems we lost travis??

**Comment by henrikstranneheim** [comment]
Coverage needs attention as well, 😄

File: lib/MIP/Recipes/Pipeline/Analyse_rd_dna_panel.pm (hunk 3 above)
Comment (jemten) [reply lib/MIP/Recipes/Pipeline/Analyse_rd_dna_panel.pm]: That bugged me as well, but I couldn't come up with a good name for the sub. Will change it for now to: 
analysis_vep -> analysis_vep_wgs
analysis_vep_rna -> analysis_vep

File: lib/MIP/Recipes/Analysis/Vep.pm [hunk 4]
```diff
@@ -1532,6 +1520,8 @@ sub _get_custom_annotation_cmds {
 
     check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
 
+    return if ( not defined $vep_custom_annotation_href );
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Vep.pm]: Nice

Comment (jemten) [reply lib/MIP/Recipes/Analysis/Vep.pm]: thanks!

**Pull request merged**